### PR TITLE
Resourcelabelevent mock

### PR DESCRIPTION
**Pull request by phdesroUbi**

### CODE REVIEW

File: NGitLab.Mock/Clients/IssueClient.cs [hunk 1]
```diff
@@ -85,12 +85,25 @@ public Models.Issue Edit(IssueEdit issueEdit)
...
+                {
+                    labelsEdit = Array.Empty<string>();
+                }
+
+                if (labelsEdit is not null)
+                {
+                    CreateResourceLabelEvents(issueToModify.Labels, labelsEdit, issueToModify.Id);
+                }
+
+                issueToModify.Labels = labelsEdit ?? issueToModify.Labels;
```
Comment (louis-z) [inline NGitLab.Mock/Clients/IssueClient.cs]: I think you can simplify the logic to something like
```suggestion
                if (!string.IsNullOrWhiteSpace(issueEdit.Labels))
                {
                    var labelsEdit = issueEdit.Labels.Split(',');
                    CreateResourceLabelEvents(issueToModify.Labels, labelsEdit, issueToModify.Id);
                    issueToModify.Labels = labelsEdit;
                }
```

File: NGitLab.Mock/Clients/IssueClient.cs [hunk 2]
```diff
@@ -361,5 +384,64 @@ private IEnumerable<Issue> FilterByQuery(IEnumerable<Issue> issues, IssueQuery q
 
             return issues;
         }
+
+        public void CreateResourceLabelEvents(string[] previousLabels, string[] newLabels, int resourceId)
```
Comment (louis-z) [inline NGitLab.Mock/Clients/IssueClient.cs]: This method can be private.

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab.Mock/Clients/IssueClient.cs (hunk 1 above)
Comment (phdesroUbi) [reply NGitLab.Mock/Clients/IssueClient.cs]: It's not the same behavior if it's null or if it's an empty string.

Comment (louis-z) [reply NGitLab.Mock/Clients/IssueClient.cs]: Ok, but I'm sure you can still simplify and/or make the logic easier to understand.

**Review by louis-z — APPROVED** [review]

**Pull request merged**